### PR TITLE
fix(provider): Change payload + allow http legacy SSL renegotiation

### PR DESCRIPTION
**Pull request by RahulDutta007**
### What change does this PR introduce?

1. Changed Gupshup (Provider) BASE URL
2. Changed provider request "Content-Type" to accept "application/json"
3. Added httpsAgent to allow legacy SSL renegoation for Gupshup API endpoint

### Why was this change needed?

1. To allow delivery of Gupshup (provider) SMS

### CODE REVIEW

File: providers/gupshup/src/lib/gupshup.provider.ts [hunk 1]
```diff
@@ -43,16 +46,14 @@ export class GupshupSmsProvider implements ISmsProvider {
...
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify(params),
-    };
+    const response = await axios.post(GupshupSmsProvider.BASE_URL, params, {
+      httpsAgent: new Agent({
+        rejectUnauthorized: false,
+        secureOptions: crypto.constants.SSL_OP_LEGACY_SERVER_CONNECT,
```
Comment (BiswaViraj) [inline providers/gupshup/src/lib/gupshup.provider.ts:52]: @RahulDutta007,
Why is this change required?

cc: @LetItRock

Comment (RahulDutta007) [reply providers/gupshup/src/lib/gupshup.provider.ts:52]: I was getting the fetch error "ERR_SSL_UNSAFE_LEGACY_RENEGOTIATION_DISABLED". Possibly due to running it as a localhost as from Node v18, it does not allow legacy TLS renegotiation and hence, unable to test.

Comment (BiswaViraj) [reply providers/gupshup/src/lib/gupshup.provider.ts:52]: @RahulDutta007, can you try using`127.0.0.1` instead of `localhost`?
I think we wont need this change.

**Comment by RahulDutta007** [comment]
Let me keep this as a simple axis request without using httsAgent. Works
for you?

**Comment by BiswaViraj** [comment]
> Let me keep this as a simple axis request without using httsAgent. Works for you?
Yes, that will work.
